packages/runtime/container-runtime/src/garbageCollectionHelpers.ts

```diff
@@ -35,3 +38,19 @@ export function sendGCUnexpectedUsageEvent(
 
 	mc.logger.sendTelemetryEvent(event, error);
 }
+
+/**
+ * In order to protect old documents that were created at a time when known bugs exist that violate GC's invariants
+ * such that enforcing GC (Fail on Tombstone load/usage, GC Sweep) would cause legitimate data loss,
+ * the container author may increment the feature support value for Tombstone such that containers created
+ * with a different value will not be subjected to GC enforcement.
+ * @param persistedValue - The persisted feature support value
+ * @param currentValue - The current app-provided feature support value
+ * @returns true if GC Enforcement (Fail on Tombstone load/usage, GC Sweep) should be disabled
+ */
+export function shouldDisableGcEnforcement(
+	persistedValue: number | undefined,
+	currentValue: number | undefined,
+): boolean {
+	return persistedValue !== currentValue;
```

Comment:
This is fine for now, but later I think we should change `undefined` persisted value to be mapped to `0` first, then compare with the incoming current value from the option. See https://dev.azure.com/fluidframework/internal/_workitems/edit/3424 for more details.

Reply:
Not sure I follow what's described in 3424 but why not return false if persistedValue is undefined? Before applications start passing the GC option, this would be undefined and tombstone will be enabled right?

Reply:
Closing this thread, let's use the other one from the test file.